### ISSUE TITLE
Grid gets disconfigured after lock screen

### ISSUE THREAD

**Issue by enriquemorenotent** [zakkak/workspace-grid#112]
I have my grid set up as a 3x3 with Ubuntu 18.04.

If I go into my lock screen, and then back to my desktrop, the grid seems to have misconfigured, and it looks like a 1x9.

I have to disable and re-enable extensions manually to restore the layout back to 3x3.

![image](https://user-images.githubusercontent.com/368842/51634560-d1dac480-1f54-11e9-82ea-61e3b2fed04a.png)

**Comment by Llbe** [comment]
Related to #111 I think.

I too have the exact same issue (since yesterday). Disabling all extensions (i.e. Tweaks → Extensions → Off → On) reinstates the grid and windows. Sometimes I need to turn off → on twice.

**Comment by Llbe** [comment]
My version of Gnome Shell is 3.28.3. I have 36 workspaces.

**Comment by Llbe** [comment]
I think this issue might be related to the upgrade process only (of Gnome Shell).

Yesterday, something crashed causing all my windows to disappear and basically nothing was responsive. I switched to terminal and TERM killed gnome-shell and restarted it. All windows returned and the extension hasn't crashed since.

TL;DR - restart gnome-shell after upgrading to 3.28.3 and the issue is solved.

**Comment by enriquemorenotent** [comment]
@Llbe Could you please share what command exactly did you use to restart gnome-shell?

**Comment by Llbe** [comment]
@enriquemorenotent: I simply found the pid of the process via `ps axu` and used `kill`. Then I ran `gnome-shell`.

If I'm correct that the issue is related to upgrading, any method of restarting gnome-shell should work.

Nothing has crashed for me since I wrote my previous message here.

**Comment by dsnt02518** [comment]
I see the same issue on CentOS 7.4, gnome-shell-3.22.3-17. Often (but not always) after a screen lock/unlock cycle grid is disabled and workspace switcher reverts to single column layout.

I also find that restarting gnome-shell fixes the issue (Alt-F2, "gnome-shell -r"). Intending to upgrade this system soon, as I also run on Ubuntu 18.04 (3.28.3-0ubuntu0.18.04.4) and don't see this issue, so I assume it's a gnome-shell version issue.

It is probably worth mentioning though that on Ubuntu/3.28 I do see the following warnings logged:

```
Feb  1 08:25:42 unifaun gnome-shell[2138]: Some code accessed the property 'refreshPropertyOnProxy' on the module 'util'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Feb  1 08:26:41 unifaun gnome-shell[2138]: Some code accessed the property 'gridWorkspaceSwitcherPopup' on the module 'gridWorkspaceSwitcherPopup'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
```
